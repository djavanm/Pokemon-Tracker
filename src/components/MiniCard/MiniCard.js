import React from 'react';

export const MiniCard = ({ pokemon, showPokemon }) => {
  const types = pokemon.types.map((type, index) => <li className={type.toLowerCase()} key={index}>{type}</li>);
  return (
    <div className='mini-card'>
    <img className='mini-gif' src={`${pokemon.gif}`} alt={`${pokemon.name}'s sprite`} onClick={(e) => showPokemon(e, pokemon.id)}/>
    <ul>
    {types}
    </ul>
    </div>
  )
};

export default MiniCard;
