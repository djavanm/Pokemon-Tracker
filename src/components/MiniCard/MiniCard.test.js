import React from 'react';
import { shallow } from 'enzyme';
import { MiniCard } from './MiniCard';


describe('Nav', () => {
  let wrapper;
  const mockShowPokemon = jest.fn();
  const mockPokemon = {
    name: 'Bulbasaur',
    description: 'Bulbasaur is cool.',
    gen: 'Generation 1',
    gif: 'https://projectpokemon.org/images/normal-sprite/bulbasaur.gif',
    height: 7,
    id: 1,
    image: 'https://assets.pokemon.com/assets/cms2/img/pokedex/full/001.png',
    sprite: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png',
    types: ['Grass', 'Poison']
  };

  beforeEach(() => {
    wrapper = shallow(<MiniCard
      showPokemon={mockShowPokemon}
      pokemon={mockPokemon}
      />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the showPokemon function when an image clicked', () => {
    wrapper.find('img').simulate('click');
    expect(mockShowPokemon).toHaveBeenCalled();
  });
});
