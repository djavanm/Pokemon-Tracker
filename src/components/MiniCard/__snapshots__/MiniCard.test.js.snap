// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
<div
  className="mini-card"
>
  <img
    alt="Bulbasaur's sprite"
    className="mini-gif"
    onClick={[Function]}
    src="https://projectpokemon.org/images/normal-sprite/bulbasaur.gif"
  />
  <ul>
    <li
      className="grass"
      key="0"
    >
      Grass
    </li>
    <li
      className="poison"
      key="1"
    >
      Poison
    </li>
  </ul>
</div>
`;
